### DISCUSSION TITLE
error with running anonymized dns

### DISCUSSION THREAD

**Discussion by t0ena1l** [DNSCrypt/dnscrypt-proxy#2623] [Potential issues]
Using linux mintOS gnome shell.

This is my first time setting up an encrypted DNS and been trying to set the anonymized DNS feature. I configured the file to my liking though when I run a command, for example `./dnscrypt-proxy -help`, it returns with an error saying `Segmentation fault (core dumped)`. I tried looking online but no luck, I also tried [this website](https://opstree.com/blog/2019/04/02/resolving-segmentation-fault-core-dumped-in-ubuntu/) but still no luck. Here's some system error logs I received, if they provide any helpful context:
```
[ 6554.380678] dnscrypt-proxy[16189]: segfault at 0 ip 0000000000466944 sp 00007ffe684c0860 error 4 in dnscrypt-proxy[400000+3e1000]
User systemd[1]: Started Process Core Dump (PID 16190/UID 0).
User systemd-coredump[16191]: Process 16189 (dnscrypt-proxy) of user 0 dumped core.#012#012Found module /home/ittos/dragonflybsd-amd64/dnscrypt-proxy without build-id.#012Found module linux-vdso.so.1 with build-id: 8fbfe39140ecade330ec9873f333f4f4200e5e10#012Stack trace of thread 16189:#012#0  0x0000000000466944 n/a (/home/ittos/dragonflybsd-amd64/dnscrypt-proxy + 0x66944)
```
Also can anyone provide some commands that checks to ensure dns is running, where its running from, etc? Thanks.

**Comment by lifenjoiner** [comment]
> linux mint
> Found module /home/ittos/dragonflybsd-amd64/dnscrypt-proxy

I'm not Linux user. But, are you using the proper binary?

**Reply by t0ena1l** [reply]
Yea I'm sure I used the correct one but maybe I can try different packages later on.

**Comment by jedisct1** [comment]
As @lifenjoiner said, it looks like you downloaded a package for DragonflyBSD, which is a different operating system. You need to download one for Linux.

**Reply by t0ena1l** [reply]
Yea, I downloaded dnscrypt-proxy-linux_x86_64-2.1.5.tar.gz which should support my system as other software binaries worked but later I can probably try downloading different ones see which work.

**Reply by jedisct1** [reply]
What does the following command print?

```sh
sha1sum /home/ittos/dragonflybsd-amd64/dnscrypt-proxy
```

**Reply by t0ena1l** [reply]
Did it print the hash?:
`a39f8d16f41698f4afad96ad5945a9846d2f20d2  /home/ittos/dragonflybsd-amd64/dnscrypt-proxy`
